### PR TITLE
chore(@kadena/react-ui): Box, Grid, Stack token refactor

### PR DESCRIPTION
**Pull request by eileenmguo**
Updated the Box, Grid, and Stack component to use `atoms` instead of `sprinkles`. 

Additionally, the Stack and Grid components are now wrapping the Box component and altering/omitting specific properties. This ensures consistency across the layout components in terms of what props are exposed.

Additionally we have decided to incrementally move to `layout` properties starting with margin and padding. 
For example: 
marginTop -> marginBlockStart
marginBottom -> marginBlockEnd
marginLeft -> marginInlineStart
marginRight -> marginInlineEnd
marginX -> marginInline
marginY -> marginBlock

CSS is moving towards logical properties which think in logical flow instead of physical position. This allows us to write less code when styling for languages that have a different writing modes (left to right vs right to left). There are many other logical properties, however it is not widely adopted and there are not logical properties for all previous physical properties. For this reason, we are staring with margin and padding so that we start thinking in terms of logical properties, and we can incrementally (or decide a complete move) to logical properties in the future 
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values

### CODE REVIEW

**Review by KristinaSpasevska — APPROVED** [review]

File: packages/apps/docs/src/components/DocsCard/DocsCard.tsx [hunk 1]
```diff
@@ -36,7 +37,12 @@ export const DocsCard: FC<IProps> = ({
       <Heading as="h3" variant="h5">
         {label}
       </Heading>
-      <Box marginY="$4" marginRight={{ sm: '$20', md: '$4', lg: '$20' }}>
+      <Box
+        marginBlock="md"
+        className={sprinkles({
+          marginInlineEnd: { sm: '$20', md: '$md', lg: '$20' },
```
Comment (eileenmguo) [inline packages/apps/docs/src/components/DocsCard/DocsCard.tsx:43]: @sstraatemans Does this look ok on the docs?

File: packages/apps/docs/src/components/Layout/Landing/components/Headers/HomeHeader.tsx [hunk 2]
```diff
@@ -77,8 +78,15 @@ export const HomeHeader: FC<IProps> = ({ popularPages }) => {
             <GridItem>
               {popularPages.length > 0 && (
                 <Box
-                  paddingLeft={{ sm: '$1', lg: '$15', xl: '$32', xxl: '$48' }}
-                  marginRight="$10"
+                  className={sprinkles({
+                    paddingInlineStart: {
+                      sm: '$xs',
```
Comment (eileenmguo) [inline packages/apps/docs/src/components/Layout/Landing/components/Headers/HomeHeader.tsx:83]: @sstraatemans this is slightly different

File: packages/apps/docs/src/pages/pact/index.tsx [hunk 3]
```diff
@@ -71,8 +74,13 @@ const Home: FC<IProps> = ({ popularPages, blogPosts }) => {
           </GridItem>
         </Grid>
 
-        <Box marginBottom="$20">
-          <Grid gap="$lg" columns={{ sm: 1, lg: 2 }}>
+        <Box
+          className={sprinkles({
+            marginBlockStart: '$8',
+            marginBlockEnd: '$20',
+          })}
```
Comment (eileenmguo) [inline packages/apps/docs/src/pages/pact/index.tsx:81]: @sstraatemans I think you added this later?

**Review by sstraatemans — APPROVED** [review]

**Review by MRVDH — APPROVED** [review]

**Review by sanderlooijenga — APPROVED** [review]

**Review by nil-amrutlal-dept — APPROVED** [review]

**Review by alber70g — APPROVED** [review]
I reviewed the files assigned to me by CODEOWNERS

**Pull request merged**